src/main/java/au/org/emii/ncdfgenerator/CreateWritable.java

```diff
@@ -2,29 +2,44 @@
 package au.org.emii.ncdfgenerator;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.lang.Thread;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 
 import ucar.nc2.NetcdfFileWriteable;
 
 
 class CreateWritable implements ICreateWritable
 {
-	// an abstraction to support creating the ncf file
 	// NetcdfFileWriteable is not an abstraction over a stream!. instead it insists on being a file...
+	// Makes resource handling tedious to support multithreading and http concurrency.
 
-	CreateWritable( String path )
+	final String tmpDir;
+
+	CreateWritable( String tmpDir )
```

Comment:
There's a bunch of broken coding conventions :smile:   

* opening brace on a newline instead of at end of method/class dec
* spaces around method params
* tabs instead of spaces